Source/portal.cpp

```diff
@@ -52,12 +52,13 @@ void AddWarpMissile(int i, int x, int y)
 	MissileData[MIS_TOWN].mlSFX = SFX_NONE;
 	dMissile[x][y] = 0;
 	mi = AddMissile({ 0, 0 }, { x, y }, 0, MIS_TOWN, TARGET_MONSTERS, i, 0, 0);
+	auto &missile = Missiles[mi];
 
 	if (mi != -1) {

```

Comment:
First make sure that `mi` is valid.
```suggestion
	if (mi != -1) {
	auto &missile = Missiles[mi];

```